### PR TITLE
MWPW-151456: Spectrum 2 Token Integration Demo

### PR DESCRIPTION
**Pull request by saashapina**
- Creates Spectrum 2 Integration Demo Page. This work includes token integration and related style adjustments for:
    - `Marquee` 
    - `Accordion`
    - `Aside`
       - adds `mobile-row-reverse` option: applies reverse order on mobile blocks  
    - `Editorial card`
        - adds `max-width` option: applies 376px max-width to editorial card
    - `Icon block`
       - adds black background to dark mode
    - `Footer`
      
screencapture-localhost-3000-drafts-ramuntea-spectrum-showcase-2024-08-02-14_47_27 (1)

Resolves: [MWPW-151456](https://jira.corp.adobe.com/browse/MWPW-151456)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://saasha-mwpw-151456-spectrum2-demo--milo--adobecom.hlx.page/drafts/ramuntea/spectrum-showcase

### CODE REVIEW

File: libs/blocks/editorial-card/editorial-card.css
Comment (overmyheadandbody) [inline libs/blocks/editorial-card/editorial-card.css:60]:
```suggestion
  margin: auto;
```

File: libs/blocks/editorial-card/editorial-card.css
Comment (overmyheadandbody) [inline libs/blocks/editorial-card/editorial-card.css:58]: I'm guessing this comes from a variation added through Sharepoint? The naming might be a bit misleading, but it should be acceptable for the purposes of the showcase page. Could you add a comment that the variant name might need to be updated?

File: libs/blocks/footer/footer.css
Comment (overmyheadandbody) [inline libs/blocks/footer/footer.css:309]: This file is used just for older pages, are you sure it achieves your desired result? It might be that we need to make some metadata changes to the page if that's the case. Same comment for L15

File: libs/blocks/icon-block/icon-block.css
Comment (overmyheadandbody) [inline libs/blocks/icon-block/icon-block.css:13]:
```suggestion
  border-radius: var(--s2-corner-radius-900);
```

File: libs/blocks/icon-block/icon-block.css
Comment (overmyheadandbody) [inline libs/blocks/icon-block/icon-block.css:11]: Why is this reduction needed for `.dark` variants?

File: libs/blocks/icon-block/icon-block.css
Comment (overmyheadandbody) [inline libs/blocks/icon-block/icon-block.css:123]:
```suggestion
  min-width: var(--icon-size-xs);
```

File: libs/blocks/marquee/marquee.css
Comment (overmyheadandbody) [inline libs/blocks/marquee/marquee.css:194]:
```suggestion
  background: var(--s2-background-layer-1-color);
```

File: libs/blocks/marquee/marquee.css
Comment (overmyheadandbody) [inline libs/blocks/marquee/marquee.css:334]:
```suggestion
  margin-bottom: var(--spacing-l);
```

File: libs/blocks/marquee/marquee.css
Comment (overmyheadandbody) [inline libs/blocks/marquee/marquee.css:329]: Do we no longer need the `order` at all?

File: libs/blocks/marquee/marquee.css
Comment (overmyheadandbody) [inline libs/blocks/marquee/marquee.css:574]: Is this intentional or forgotten after some testing? Just asking because we usually have variables for colors.

File: libs/styles/styles.css
Comment (overmyheadandbody) [inline libs/styles/styles.css:22]: Is the color no longer available from Spectrum?

File: libs/styles/styles.css
Comment (overmyheadandbody) [inline libs/styles/styles.css:406]:
```suggestion
    --grid-container-width: 1600px;
```

File: libs/styles/styles.css
Comment (overmyheadandbody) [inline libs/styles/styles.css:1079]:
```suggestion
}
```

File: libs/styles/styles.css
Comment (overmyheadandbody) [inline libs/styles/styles.css:1073]: Why do we need to completely remove the `.light` styles? It might be that they are still valuable for backwards compatibility across our pages.

**Review by overmyheadandbody — COMMENTED** [review]
A few nits and questions, but overall this looks good, great job!

I'll also dive into a visual inspection soon. The first thing that instantly popped up was the footer where the links seem a bit too close for comfort. It looks like it has to do with some `line-height` definitions
Screenshot 2024-08-05 at 14 39 53

**Comment by overmyheadandbody** [comment]
As for the quick visual inspection:

### Marquee
* something is off with the Adobe Photoshop logo and its relationship with the Marquee title. It might be the logo size or some spacing between the two

### Aside
* the title seems to be `heading-xl`, but the Figma file seems to indicate size L should be used. The detail is M-sized, while Figma suggests L

### Editorial Cards
* these might need a bit more attention - the cell padding, the title size and the link color are different compared to the Figma file

| Demo Page | Figma |
| ------------- | ------------- |
| Screenshot 2024-08-05 at 15 10 40 | Screenshot 2024-08-05 at 15 11 08 |

### FAQ
* the Figma file seems to suggest a wider container of 1200px, while currently it's set at 1000px
* I don't remember where we landed on link colors 😅 Figma seems to suggest blue links, while we're using ones of the same color as the text. Am I remembering correctly that we said this is a current limitation?

### Icon Block
* I'm still not convinced what the intention with its width is, we should ask this question

### Footer
* I already brought this up, but I see your screenshot has the correct styles. I also tried the link in incognito, but get the same compact style. We can sync on this if needed to see what the issue might be

File: libs/blocks/aside/aside.css
Comment (saashapina) [reply libs/blocks/aside/aside.css:638]: Yea that's a good point. The reason this was removed was because the spectrum mockup specifies these new aside large blocks to be 640px in height instead of 700px. Removing this allows the height to fall to the height of the blocks content. However, we could consider explicitly setting a min-height of 640px here instead, which might help mitigate the impact on existing production use cases.

File: libs/blocks/editorial-card/editorial-card.css
Comment (saashapina) [reply libs/blocks/editorial-card/editorial-card.css:58]: Yes it is. Yea I wasn't sure what the best variation name should be for this but I'm open to suggestions to make this more clear. I'll add in that comment for now.

File: libs/blocks/footer/footer.css
Comment (saashapina) [reply libs/blocks/footer/footer.css:309]: Good catch! It seems I was experiencing a rendering issue locally, where this original file for the footer was being displayed instead of the global footer. That would explain why I wasn't seeing the line height issue I thought I fixed 🙃. Restarting my server looked like it fixed the issue.

File: libs/blocks/marquee/marquee.css
Comment (saashapina) [reply libs/blocks/marquee/marquee.css:574]: Good catch, yes this was from a test.

File: libs/blocks/marquee/marquee.css
Comment (saashapina) [reply libs/blocks/marquee/marquee.css:329]: No, I don't believe we need this property any more. It was previously used to position the action CTA buttons above the body text on mobile devices but the new mockups are indicating that the buttons should now fall in their original order.

File: libs/styles/styles.css
Comment (saashapina) [reply libs/styles/styles.css:1073]: I can add these styles back in for backwards compatibility and remove the light properties from the blocks here instead. I initially had these removed because the spectrum designs were created to default to the `light` styles automatically, so it's only needed to toggle to `dark` when needed.

File: libs/styles/styles.css
Comment (saashapina) [reply libs/styles/styles.css:22]: Yes it is, there was a conflict in the designs I noticed when I was seeing the old footer background color using the `gray-100` s2 token color. It was a darker gray then specified in our mockups and the mockups called for this original color gray to still be used. I kept both this milo gray-100 and the s2 gray-100 in temporarily to see if it needed to be discussed with design, but that issue is actually fixed with correct usage of the global footer using the `--feds-background-footer--light`

**Comment by saashapina** [comment]
> A few nits and questions, but overall this looks good, great job!
> 
> I'll also dive into a visual inspection soon. The first thing that instantly popped up was the footer where the links seem a bit too close for comfort. It looks like it has to do with some `line-height` definitions Screenshot 2024-08-05 at 14 39 53

Figured out the issue with the footer was because there was some issue causing the old footer to render for me locally instead of the global. This issue should be fixed now that the changes are added to the intended area.

File: libs/blocks/icon-block/icon-block.css
Comment (saashapina) [reply libs/blocks/icon-block/icon-block.css:11]: This width reduction is needed to accommodate the black background on the `.dark` variant that is "full-width" but not entirely full-width per the mockup.
Screenshot 2024-08-06 at 12 08 10 AM

File: libs/blocks/global-footer/global-footer.css
Comment (narcis-radu) [inline libs/blocks/global-footer/global-footer.css:225]: Why is the font size different for the copyright section?

**Review by narcis-radu — COMMENTED** [review]
I noticed a couple of small differences and I'm not sure whether it's because of the design change or just some inconsistencies.
1. The editorial cards are not really aligned with the text above anymore
Screenshot 2024-08-06 at 15 15 50

2. FAQ links have the same color as text and there's no style update on hover
Screenshot 2024-08-06 at 15 17 03

3. Region selector links have the same color as normal text and hovering does not produce a color change
Screenshot 2024-08-06 at 15 17 59

File: libs/blocks/aside/aside.css
Comment (overmyheadandbody) [reply libs/blocks/aside/aside.css]: Yep, setting it to `640` is likely the better definition to ensure current marquees aren't impacted too much.

File: libs/blocks/accordion/accordion.css
Comment (overmyheadandbody) [inline libs/blocks/accordion/accordion.css:25]: Do we have a need for the fallbacks at this point?

**Review by overmyheadandbody — COMMENTED** [review]
Thanks for integrating all the feedback! I see @narcis-radu also has a few questions worth following up on.

Looking at the design side of things, I still see a few gaps:
* Aside heading size is still off, it now has both `l` and `xl` heading classes, we might have to look into that;
* Editorial cards are looking better, but I feel the title doesn't match the Figma specs. I've tried to replace the current `xs` size with an `s`, but then it gets too large, have all heading font-sizes been updated?
* for the FAQ section, I just wonder about the inner links' color

File: libs/blocks/global-footer/global-footer.css
Comment (saashapina) [reply libs/blocks/global-footer/global-footer.css:225]: This was a change specified in the mockups, but we can revert this if its not needed.
Screenshot 2024-08-06 at 10 12 36 AM

Comment (narcis-radu) [reply libs/blocks/global-footer/global-footer.css:225]: @saashapina - is the other content in the footer 11px as well?

**Comment by saashapina** [comment]
> Thanks for integrating all the feedback! I see @narcis-radu also has a few questions worth following up on.
> 
> Looking at the design side of things, I still see a few gaps:
> 
> * Aside heading size is still off, it now has both `l` and `xl` heading classes, we might have to look into that;
> * Editorial cards are looking better, but I feel the title doesn't match the Figma specs. I've tried to replace the current `xs` size with an `s`, but then it gets too large, have all heading font-sizes been updated?
> * for the FAQ section, I just wonder about the inner links' color

Thanks for catching those details.

- The link color issue in the FAQ section should now be addressed. Looked like there was a typo in the variables causing this.
- Regarding the editorial card heading, yes, all heading font sizes have been updated. However, the mockups specify using the new `Title XL` font size instead of the standard heading sizes for the editorial cards, which is likely why the xs or s heading sizes didn't work for those. It has now been corrected to use the title styles.
- The aside heading issue has also been addressed.

Another quick note regarding the editorial cards is the multi-line truncated paragraph with ellipses. CSS currently lacks a native property for this effect, and while we could use JS, it might have performance implications. It could be worth discussing this with the design team. The best solution might be to create a shortened version of the description and add a hard-coded ellipsis when necessary.

File: libs/blocks/global-footer/global-footer.css
Comment (saashapina) [reply libs/blocks/global-footer/global-footer.css:225]: No, it looks like the other content in the footer remains the same as the original at 12px. It was only the details in this privacy section that was updated to the 11px.

**Comment by saashapina** [comment]
> As for the quick visual inspection:
> 
> ### Marquee
> * something is off with the Adobe Photoshop logo and its relationship with the Marquee title. It might be the logo size or some spacing between the two
> 


Regarding the Marquee Adobe logo, it was challenging to match the exact specifications since the mockup uses text while we were provided an image. The current spacing is based on the variables specified in the mockup. We may need to adjust the logo size further, but this is the closest match I could achieve using the available Spectrum icon variables. Let me know if you have any other suggestions for this.

**Comment by overmyheadandbody** [comment]
Re: editorial cards ellipsis, you should be able to do
```
display: -webkit-box;
-webkit-line-clamp: 2;
-webkit-box-orient: vertical;
overflow: hidden;
text-overflow: ellipsis;
```
and this is the result:
Screenshot 2024-08-07 at 13 21 40

File: libs/styles/styles.css
Comment (overmyheadandbody) [inline libs/styles/styles.css:1382]: I'm not very convinced with removing this, it adds underline for links in the bottom part of the footer and the region modal, maybe we can adapt this closer to where we actually need it, like the FAQ / Editorial Card sections.

**Review by overmyheadandbody — COMMENTED** [review]
Just one more nit. And I see the top navigation is now missing, it looks like a metadata issue, I'll look into this tomorrow and fix it.

Otherwise I think we're close to sending this to QA for another round of visual inspection and then we can bring it up to the stakeholders. Great job!

File: libs/styles/styles.css
Comment (saashapina) [reply libs/styles/styles.css:1382]: Yea the reason I removed this is because the Spectrum design system specified that the links should be underlined by default. It seems like the footer, bottom footer, and the region modal are special cases where it shouldn't be applied according to the mockups. It made sense to handle these specific cases individually rather than overriding the default styling globally. What do you think about this approach? Or do you think we should only add the underline to the FAQ/ Editorial cards for the purposes of the demo page?
Screenshot 2024-08-07 at 11 29 56 PM

File: libs/blocks/global-footer/global-footer.css
Comment (narcis-radu) [reply libs/blocks/global-footer/global-footer.css:226]: I did some investigations and looks like the copyright text was 11px in the AEM Global Navigation, meaning the change is correct. I will mark this as resolved, no further action needed. Thanks!

File: libs/styles/styles.css
Comment (overmyheadandbody) [reply libs/styles/styles.css:1382]: If the new design decision is to underline, then I think what you're suggesting is better for now, remove text decoration where it's not needed.

File: libs/blocks/editorial-card/editorial-card.js
Comment (overmyheadandbody) [inline libs/blocks/editorial-card/editorial-card.js:57]: Could we add a `// TODO` comment with some details to make it clear that this needs to be revisited?

**Review by overmyheadandbody — COMMENTED** [review]
I was now looking over the mobile design as well and I have a few notes:
* Marquee CTAs have higher height in the Figma file compared to the implementation;
* for the editorial cards section we need to communicate we can't use a slider for the cards right now; additionally, I see the CTA drops below the cards, that's also something we can't achieve OOTB;
* in the icon block, the CTAs seem to stack and are wider and taller in the Figma file compared to the implementation;

Also a few notes for tablets:
* same CTA height note as for mobile;
* there seems to be more space between asides than expected:
Screenshot 2024-08-08 at 15 37 19

* same editorial cards comment, we'll have to communicate we can't achieve that layout right now without additional work;

I'm hoping this is the last round of feedback 😅  It's a big PR with a lot of moving parts, sorry for all the back and forth!

**Comment by overmyheadandbody** [comment]
Also, I had a look over the Gnav issue, it seems it's related to your branch, the `global-navigation/base.css` file is not updated on your branch, things should be working as expected as soon as we merge this into the feature branch. Alternatively, you could try to rebase the feature branch with the current `stage` one, then rebase your changes to the feature branch after pulling it.

**Review by overmyheadandbody — APPROVED** [review]
Thanks for integrating all the feedback! I think this is ready for a round of QA and then we can provide details for the stakeholders on how to access this page for showcase purposes.

Pending a second approval, I'd like to merge this in the feature branch so QA can do a sort of E2E testing.

**Review by mokimo — APPROVED** [review]

**Pull request merged**
